core/trino-main/src/main/java/io/trino/execution/scheduler/TaskExecutionClass.java

```diff
@@ -30,4 +30,12 @@ boolean canTransitionTo(TaskExecutionClass targetExecutionClass)
             case SPECULATIVE -> targetExecutionClass == SPECULATIVE || targetExecutionClass == STANDARD;
         };
     }
+
+    boolean isSpeculative()
```

Comment:
> Add helper isSpeculative method

why? do we envision new options?
maybe add a commti mesage

Reply:
Yeah. Added commit message.